### PR TITLE
sys-process/below: new ebuild

### PR DESCRIPTION
**Pull request by hodgesds**
Signed-off-by: Daniel Hodges <hodges.daniel.scott@gmail.com>

### CODE REVIEW

File: sys-process/below/below-0.2.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+RESTRICT="mirror"
+# License set may be more restrictive as OR is not respected
+# use cargo-license for a more accurate license picture
+LICENSE="Apache-2.0 Apache-2.0 WITH LLVM-exception BSD-2-Clause BSD-3-Clause BSL-1.0 ISC LGPL-2.1 MIT MPL-2.0 Unlicense"
+SLOT="0"
+KEYWORDS="~amd64"
+IUSE=""
+
+DEPEND=""
+RDEPEND=""
```
Comment (scardracs) [inline sys-process/below/below-0.2.0.ebuild:230]: Delete these as they are empty

File: sys-process/below/metadata.xml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE pkgmetadata SYSTEM "http://www.gentoo.org/dtd/metadata.dtd">
+<pkgmetadata>
+	<maintainer type="person">
+		<email>hodges.daniel.scott@gmail.com</email>
+		<name>Daniel Hodges</name>
+	</maintainer>
```
Comment (scardracs) [inline sys-process/below/metadata.xml:7]: You aren't a dev so you have to change with:

`
	<maintainer type="person" proxied="yes">
		<email>hodges.daniel.scott@gmail.com</email>
		<name>Daniel Hodges</name>
	</maintainer>
	<maintainer type="project" proxied="proxy">
		<email>proxy-maint@gentoo.org</email>
		<name>Proxy Maintainers</name>
	</maintainer>
`

Comment (scardracs) [inline sys-process/below/metadata.xml:7]: You are not a dev so change like that https://dpaste.com/6YTZQMYAE

File: sys-process/below/below-0.2.0.ebuild [hunk 3]
```diff
@@ -0,0 +1,230 @@
+# Copyright 2017-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+# Auto-Generated by cargo-ebuild 0.3.1
+
```
Comment (scardracs) [inline sys-process/below/below-0.2.0.ebuild:5]: Delete these lines

File: sys-process/below/below-0.2.0.ebuild [hunk 4]
```diff
@@ -0,0 +1,230 @@
+# Copyright 2017-2021 Gentoo Authors
```
Comment (scardracs) [inline sys-process/below/below-0.2.0.ebuild:1]: 1999-2021 or 2021

**Comment by scardracs** [comment]
Did you run `ebuild sys-process/below-0.2.0.ebuild manifest` before uploading? The Manifest does not exist

**Comment by scardracs** [comment]
Also please change the title with sys-process/below: new ebuild

**Comment by scardracs** [comment]
After that, if this package is not required for some packages already inside gentoo, please consider adding it to https://wiki.gentoo.org/wiki/Project:GURU

**Comment by hodgesds** [comment]
Thanks for the helpful review! Will try to update within the next day.

File: sys-process/below/below-0.2.0.ebuild [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+zstd-sys-1.5.0+zstd.1.4.9
+"
+
+inherit cargo
+
+DESCRIPTION="An interactive tool to view and record historical system data"
+HOMEPAGE="https://github.com/facebookincubator/below"
+SRC_URI="$(cargo_crate_uris ${CRATES})"
+RESTRICT="mirror"
+LICENSE="Apache-2.0 Apache-2.0 WITH LLVM-exception BSD-2-Clause BSD-3-Clause BSL-1.0 ISC LGPL-2.1 MIT MPL-2.0 Unlicense"
```
Comment (Flowdalic) [inline sys-process/below/below-0.2.0.ebuild:218]: Duplicate license: Apache-2.0

Comment (hodgesds) [reply sys-process/below/below-0.2.0.ebuild:218]: Let me make an issue for [cargo-ebuild](https://github.com/cardoe/cargo-ebuild) to get the license generation fixed.

Comment (Flowdalic) [reply sys-process/below/below-0.2.0.ebuild]: Note that some the other license strings are invalid too

File: sys-process/below/below-0.2.0.ebuild [hunk 6]
```diff
@@ -0,0 +1,220 @@
...
+zstd-0.7.0+zstd.1.4.9
+zstd-safe-3.1.0+zstd.1.4.9
+zstd-sys-1.5.0+zstd.1.4.9
+"
+
+inherit cargo
+
+DESCRIPTION="An interactive tool to view and record historical system data"
+HOMEPAGE="https://github.com/facebookincubator/below"
+SRC_URI="$(cargo_crate_uris ${CRATES})"
```
Comment (Flowdalic) [inline sys-process/below/below-0.2.0.ebuild:216]: That SRC_URI is not complete. You are aware that the ebuild can't work this way? Please test your stuff before creating a pull request!

File: sys-process/below/below-0.2.0.ebuild [hunk 7]
```diff
@@ -0,0 +1,227 @@
...
+
+DESCRIPTION="An interactive tool to view and record historical system data"
+HOMEPAGE="https://github.com/facebookincubator/below"
+SRC_URI="https://github.com/facebookincubator/${PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz
+    $(cargo_crate_uris ${CRATES})"
+RESTRICT="mirror"
+LICENSE="Apache-2.0 Apache-2.0-with-LLVM-exceptions BSD-2 BSD-3 Boost-1.0 ISC LGPL-2.1 MIT MPL-2.0 Unlicense"
+SLOT="0"
+KEYWORDS="~amd64"
+COMMON_DEPEND="virtual/libelf"
```
Comment (gyakovlev) [inline sys-process/below/below-0.2.0.ebuild:222]: `COMMON_DEPEND` is not a standard variable, nothing uses it, you have to reference it, but not even in this case.
you can just specify

RDEPEND="virtual/libelf"
DEPEND="${RDEPEND}
    dev-libs/libbpf
    sys-libs/ncurses"

File: sys-process/below/below-0.2.0.ebuild [hunk 8]
```diff
@@ -0,0 +1,227 @@
...
+zstd-safe-3.1.0+zstd.1.4.9
+zstd-sys-1.5.0+zstd.1.4.9
+"
+
+inherit cargo
+
+DESCRIPTION="An interactive tool to view and record historical system data"
+HOMEPAGE="https://github.com/facebookincubator/below"
+SRC_URI="https://github.com/facebookincubator/${PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz
+    $(cargo_crate_uris ${CRATES})"
```
Comment (gyakovlev) [inline sys-process/below/below-0.2.0.ebuild:217]: you can omit passing '${CRATES}` here. just `cargo_crate_uris` is enough.

**Review by gyakovlev — CHANGES_REQUESTED** [review]

**Comment by gyakovlev** [comment]
it also needs

`QA_FLAGS_IGNORED="usr/bin/below"`

otherwise there's a warning
```
 * QA Notice: Files built without respecting LDFLAGS have been detected
 *  Please include the following list of files in your report:
 * /usr/bin/below

```

and it's missing dependency on `sys-libs/zlib`

**Comment by gyakovlev** [comment]
also needs `virtual/pkgconfig` in `BDEPEND`

File: sys-process/below/below-0.2.0.ebuild [hunk 9]
```diff
@@ -0,0 +1,227 @@
...
+RESTRICT="mirror"
```
Comment (gyakovlev) [inline sys-process/below/below-0.2.0.ebuild:218]: remove that `RESTRICT`, it's needed only for overlays.

**Comment by hodgesds** [comment]
Thanks for the helpful reviews. I still need to learn some more ebuild internals and learned to only depend on the `cargo-ebuild` tool for making templates of ebuilds.

**Pull request closed without merging**